### ISSUE TITLE
Fatal issue possibly due to vite-plugin-ssr@0.4.59

### ISSUE THREAD

**Issue by uifox** [brillout/vite-plugin-ssr#556]
### Description

After trying to upgrade from v0.4.44 to `vite-plugin-ssr@0.4.59` my app fails both in development and production mode. See stack trace below.

Relevant dependencies of my app:
* node@16
* vue@3.2.25
* vite@3.2.0
* vite-plugin-ssr@0.4.59
* vite-plugin-environment@1.1.2
* vite-plugin-vuetify@1.0.1
* express@4.18.1
* vuetify@3.0.3
* typescript@4.7.4
* vue-tsc@1.0.9

The first version I can successfully upgrade to is v0.4.54.

Relevant part of my `vite.config.ts`:
```ts
export default defineConfig(({ mode }) => {
  return {
    base: 'https://www.example.com/',
    plugins: [
      ssr()
      // etc
    ]
    // other config options
    // ...
  }
})
```

### Error Message + Error Stack

```shell
root@3bef86e36412:/home/node# yarn dev
yarn run v1.22.19
$ yarn run server
$ ts-node ./server
/home/node/node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/baseUrls.js:35
            (0, utils_1.assert)(configVps.baseAssets === baseAssets);
                               ^
Error: [vite-plugin-ssr@0.4.59][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Reach out at https://github.com/brillout/vite-plugin-ssr/issues/new or https://discord.com/invite/qTq92FQzKb and include this error stack (the error stack is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust.
    at configResolved (/home/node/node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/baseUrls.js:35:32)
    at async Promise.all (index 12)
    at async resolveConfig (file:///home/node/node_modules/vite/dist/node/chunks/dep-61d2428a.js:65483:5)
    at async Module.createServer (file:///home/node/node_modules/vite/dist/node/chunks/dep-61d2428a.js:64547:20)
    at async initVite (file:///home/node/server/index.ts:95:21)
    at async startServer (file:///home/node/server/index.ts:64:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
```

**Comment by brillout** [comment]
Insert
```js
            console.log('configVps.baseAssets', configVps.baseAssets);
            console.log('baseAsssets', baseAssets);
```
before the line `node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/baseUrls.js:35` and report back.

**Comment by uifox** [comment]
> Insert
> 
> ```js
>             console.log('configVps.baseAssets', configVps.baseAssets);
>             console.log('baseAsssets', baseAssets);
> ```
> 
> before the line `node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/baseUrls.js:35` and report back.

```
configVps.baseAssets:  /
baseAssets:  http://www.balazssnuc.doxo.com/
```

**Comment by brillout** [comment]
> If you remove the failing assertion, does your app work then?

Regardless, you'll likely hit a problem sooner or later if we ignore the assertion.

We should get to the bottom of this.

Change your `vite.config.js` to be a plain JavaScript object instead of a function and see if that makes a different.

If it doesn't make a difference then it seems like one of your Vite plugins is erasing the `vite.config.js#base` value.

If you can provide a *minimal* reproduction, I'll have a look at it. (But you'll likely find the problem before you manage to create a minimal reproduction.)

**Comment by uifox** [comment]
> Change your vite.config.js to be a plain JavaScript object instead of a function and see if that makes a different.

I changed it. No difference.

> one of your Vite plugins is erasing the vite.config.js#base value

I commented out all plugins except vite-ssr, but that didn't solve the issue either. I cleaned up my vite config almost removing everything from it, yet no change. I'll try to come up with a minimal reproduction and get back to you ASAP.

Thanks a lot for your help.

**Comment by brillout** [comment]
Hm, strange. Looking forward to the min reprod.

**Comment by brillout** [comment]
You can use one of the `examples/base-url*`. Most notably the `examples/base-url-cdn` is prob a good start.

**Comment by uifox** [comment]
I commented out the asserts in [baseUrls.ts#L37-L38](https://github.com/brillout/vite-plugin-ssr/blob/609504200fa4d94708c60d9e2dae695f6fc46b74/vite-plugin-ssr/node/plugin/plugins/baseUrls.ts#L37-L38) and now it is failing in [extensionsAssets.ts#L40](https://github.com/brillout/vite-plugin-ssr/blob/696d602fa3c0f28941f94758b85f47ea92ac279e/vite-plugin-ssr/node/plugin/plugins/extensionsAssets.ts#L40).

It is because we have a custom `assetsDir`:
```js
build: {
  assetsDir: 'sidra_assets'
}
```

Using the default `assetsDir` lets me start the app and it functions as expected, however, due to how app resources are managed at the company I work for, it is preferable for us to use a custom `assetsDir`.

Is there a hard requirement in this plugin to name the `assetsDir` "assets"? Could this be eased?

Even after using the default `assetsDir` when I add back the `asserts` to baseUrls.ts, the original failure is back, so I am still working on a minimal reproduction.

**Comment by brillout** [comment]
Just released (`0.4.61`) a fix for the `assetsDir` issue. I didn't test it but it should work. Let me know if it doesn't.

> Even after using the default `assetsDir` when I add back the `asserts` to baseUrls.ts, the original failure is back, so I am still working on a minimal reproduction.

I'm looking forward to the minimal reproduction.

**Comment by uifox** [comment]
As it turns out the mere fact that I pass in the `base` option to the vite config breaks the app. e.g. `base: 'http://www.balazssnuc.doxo.com/foo/'` will result in:

```
baseAssets:  http://www.balazssnuc.doxo.com/foo/
configVps.baseAssets:  /foo/
```

If I don't define the `base` config option the app starts and the assert does not fail.
```
baseAssets:  /
configVps.baseAssets:  /
```

**Comment by uifox** [comment]
Here is a minimal reproduction:
Change `'/some/base-url/'` here: https://github.com/brillout/vite-plugin-ssr/blob/91633fbddf39461742c0cceb47791b96b59d84c0/examples/base-url/vite.config.js#L5 to an absolute URL, e.g. `'https://www.example.com/some/base-url/'`.

Error log:
```
:~/work/3rd-party/vite-plugin-ssr/examples/base-url (main)$ yarn dev
yarn run v1.22.4
warning package.json: No license field
$ vite
error when starting dev server:
Error: [vite-plugin-ssr@0.4.61][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Reach out at https://github.com/brillout/vite-plugin-ssr/issues/new or https://discord.com/invite/qTq92FQzKb and include this error stack (the error stack is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust.
    at configResolved (/Users/balazss/work/3rd-party/vite-plugin-ssr/examples/base-url/node_modules/vite-plugin-ssr/dist/cjs/node/plugin/plugins/baseUrls.js:34:32)
    at async Promise.all (index 11)
    at async resolveConfig (file:///Users/balazss/work/3rd-party/vite-plugin-ssr/examples/base-url/node_modules/vite/dist/node/chunks/dep-5605cfa4.js:62878:5)
    at async createServer (file:///Users/balazss/work/3rd-party/vite-plugin-ssr/examples/base-url/node_modules/vite/dist/node/chunks/dep-5605cfa4.js:61943:20)
    at async CAC.<anonymous> (file:///Users/balazss/work/3rd-party/vite-plugin-ssr/examples/base-url/node_modules/vite/dist/node/cli.js:707:24)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Adding following log statements to [configResolved(config)](https://github.com/brillout/vite-plugin-ssr/blob/609504200fa4d94708c60d9e2dae695f6fc46b74/vite-plugin-ssr/node/plugin/plugins/baseUrls.ts#L33):
```js
async configResolved(config) {
    const configVps = await (0, assertConfigVps_1.getConfigVps)(config);
    // Ensure that the premature base URL resolving we did in config() isn't erroneous
    console.log('configVps.baseServer: ', configVps.baseServer);
    console.log('baseServer: ', baseServer);
    console.log('configVps.baseAssets: ', configVps.baseAssets);
    console.log('baseAssets: ', baseAssets);
    console.log('config.base: ', config.base);
    
    (0, utils_1.assert)(configVps.baseServer === baseServer);
    (0, utils_1.assert)(configVps.baseAssets === baseAssets);
    (0, utils_1.assert)(config.base === baseAssets);
}
```

prints these values:
```
configVps.baseServer:  /some/base-url/
baseServer:  /
configVps.baseAssets:  /some/base-url/
baseAssets:  https://www.example.com/some/base-url/
config.base:  /some/base-url/
```

Apparently `config.base` is reassigned perhaps by vite itself?

**Comment by brillout** [comment]
Fix released in `0.4.62`. But I'm afraid Vite is buggy around setting `base` to an absolute URL in development. Do you need this for dev? CDNs are usually only used in prod.

**Comment by uifox** [comment]
> Fix released in `0.4.62`. But I'm afraid Vite is buggy around setting `base` to an absolute URL in development. Do you need this for dev? CDNs are usually only used in prod.

Used both in dev and production.
Thanks for the fix.

**Comment by brillout** [comment]
I'm glad it works out for you.

FYI https://github.com/sponsors/brillout in case your company is up for that. Any help is appreciated!
